### ISSUE TITLE
Stack Reference specific doc

### ISSUE THREAD

**Issue by sean1588** [pulumi/docs#4701]
### Problem description

We are currently linking to this [doc](https://www.pulumi.com/docs/intro/concepts/organizing-stacks-projects/#inter-stack-dependencies) to guide the user through stack references from the checklist that will be shown on the user landing page. I feel this doc is a bit hard to follow for users who are new to Pulumi when configuring this for the first time. We should have a doc that is caters more to users who will be configuring this for the first time. Maybe something that is more along the lines of a guide would be more appropriate for this use case? I think we should also be more consistent with our verbiage, i.e. we refer to them as inter-stack dependencies here instead of stack references.

**Comment by susanev** [comment]
@davidwrede would love your thoughts on this, whenever you have time

**Comment by davidwrede** [comment]
FYI, #4707 will contain updates to all of the current programming model docs (including stacks and stack references) and it looks like the updates to the stack conceptual content (including stack references) should be more approachable for new users.

**Comment by susanev** [comment]
im going to close this out as we've made a number of improvements since this issue was opened
